sys/include/net/nanocoap.h

```diff
@@ -390,6 +400,21 @@ ssize_t coap_handle_req(coap_pkt_t *pkt, uint8_t *resp_buf, unsigned resp_buf_le
 ssize_t coap_build_hdr(coap_hdr_t *hdr, unsigned type, uint8_t *token,
                        size_t token_len, unsigned code, uint16_t id);
 
+/**
+ * @brief   Initialize a packet struct, to build a message buffer
+ *
+ * @post pkt.flags all zeroed
+ * @post pkt.payload points to first byte after header
+ * @post pkt.payload_len set to maximum space available for options + payload
+ *
+ * @param[out]   pkt        pkt to initialize
+ * @param[in]    buf        buffer to write for pkt, with CoAP header already
+ *                          initialized
+ * @param[in]    len        length of buf
+ * @param[in]    header_len length of header in buf, including token
+ */
+void coap_pkt_init(coap_pkt_t *pkt, uint8_t *buf, size_t len, size_t header_len);
```

Comment:
I think an explicit @pre to indicate that there should be an initialized `coap_hdr_t` in the buffer would be nice. Somebody is going to miss the note with the `buf` param there :)

Reply:
I agree completely.